src/batch.js

```diff
@@ -10,6 +11,49 @@ function mountOnto(router) {
   });
 }
 
+function parseURL(URL) {
+  if (typeof URL === 'string') {
+    return url.parse(URL)
+  }
+  return undefined;
+}
+
+function mBatchRoutingPath(originalUrl, serverURL, publicServerURL) {
+    serverURL = serverURL ? parseURL(serverURL) : undefined;
```

Comment:
minor  but could you just move the check for URL being defined into parseURL? e.g. change in the function to ```if (URL && typeof URL === 'string') {```